### ISSUE TITLE
Please post training curves for Thumos

### ISSUE THREAD

**Issue by cinjon** [wzmsltw/BSN-boundary-sensitive-network.pytorch#19]
Here are what mine look like using the provided code for Thumos dataset, but converted to Pytorch. I am trying to debug this because I don't think these are correct given the poor test results. 

Training curve. The line to the right is the same as the line to the left, except using 4 gpus and a corresponding 4x batch size and 4x learning rate.
![train_total_loss-VS-step](https://user-images.githubusercontent.com/615351/64642109-9acdfc00-d3db-11e9-945e-3e7ea14e4d6b.jpeg)

Test curve. Note how this never goes down; it only shows overfitting to Train.
![test_total_loss-VS-step](https://user-images.githubusercontent.com/615351/64642108-9acdfc00-d3db-11e9-8843-0669a830af06.jpeg)

**Comment by cinjon** [comment]
After running the TF loop, these are the losses. While train loss is going down, there appears to be a lot of overfitting (see Val Cost/Loss below), although not to the extent that I am seeing in the above graphs. 

Can you comment on this? Is this expected?

Epoch-0 Train Loss: Total - 3.3631, Cost - 2.1254, L2 - 1.2377, Action - 0.48, Start - 0.60, End - 0.57
Epoch-0 Val Loss: Total - 2.8303, Cost - 2.2215, L2 - 0.6088, Action - 0.50, Start - 0.62, End - 0.60
Epoch-1 Train Loss: Total - 2.3602, Cost - 1.9053, L2 - 0.4549, Action - 0.43, Start - 0.54, End - 0.51
Epoch-1 Val Loss: Total - 2.5805, Cost - 2.2222, L2 - 0.3583, Action - 0.51, Start - 0.63, End - 0.59
Epoch-2 Train Loss: Total - 2.1259, Cost - 1.8070, L2 - 0.3190, Action - 0.40, Start - 0.51, End - 0.49
Epoch-2 Val Loss: Total - 2.5622, Cost - 2.2704, L2 - 0.2917, Action - 0.51, Start - 0.63, End - 0.61
Epoch-3 Train Loss: Total - 2.0468, Cost - 1.7611, L2 - 0.2857, Action - 0.39, Start - 0.50, End - 0.48
Epoch-3 Val Loss: Total - 2.6030, Cost - 2.3270, L2 - 0.2759, Action - 0.54, Start - 0.63, End - 0.62
Epoch-4 Train Loss: Total - 1.9805, Cost - 1.7036, L2 - 0.2769, Action - 0.38, Start - 0.47, End - 0.46
Epoch-4 Val Loss: Total - 2.5809, Cost - 2.2990, L2 - 0.2819, Action - 0.53, Start - 0.63, End - 0.60
Epoch-5 Train Loss: Total - 1.8115, Cost - 1.5346, L2 - 0.2769, Action - 0.34, Start - 0.43, End - 0.42
Epoch-5 Val Loss: Total - 2.5173, Cost - 2.2457, L2 - 0.2716, Action - 0.51, Start - 0.63, End - 0.60
Epoch-6 Train Loss: Total - 1.7202, Cost - 1.4521, L2 - 0.2681, Action - 0.32, Start - 0.41, End - 0.40
Epoch-6 Val Loss: Total - 2.5561, Cost - 2.2914, L2 - 0.2647, Action - 0.51, Start - 0.66, End - 0.60
Epoch-7 Train Loss: Total - 1.6736, Cost - 1.4109, L2 - 0.2627, Action - 0.32, Start - 0.40, End - 0.38
Epoch-7 Val Loss: Total - 2.5702, Cost - 2.3095, L2 - 0.2607, Action - 0.52, Start - 0.67, End - 0.61
Epoch-8 Train Loss: Total - 1.6425, Cost - 1.3826, L2 - 0.2599, Action - 0.31, Start - 0.39, End - 0.38
Epoch-8 Val Loss: Total - 2.6026, Cost - 2.3437, L2 - 0.2590, Action - 0.52, Start - 0.69, End - 0.62
Epoch-9 Train Loss: Total - 1.6052, Cost - 1.3460, L2 - 0.2592, Action - 0.30, Start - 0.38, End - 0.37
Epoch-9 Val Loss: Total - 2.6492, Cost - 2.3903, L2 - 0.2589, Action - 0.53, Start - 0.69, End - 0.64
Epoch-10 Train Loss: Total - 1.5805, Cost - 1.3208, L2 - 0.2597, Action - 0.30, Start - 0.37, End - 0.36
Epoch-10 Val Loss: Total - 2.6554, Cost - 2.3954, L2 - 0.2600, Action - 0.54, Start - 0.69, End - 0.64
Epoch-11 Train Loss: Total - 1.5588, Cost - 1.2976, L2 - 0.2612, Action - 0.29, Start - 0.36, End - 0.35
Epoch-11 Val Loss: Total - 2.6561, Cost - 2.3942, L2 - 0.2619, Action - 0.53, Start - 0.70, End - 0.63
Epoch-12 Train Loss: Total - 1.5274, Cost - 1.2641, L2 - 0.2634, Action - 0.28, Start - 0.35, End - 0.35
Epoch-12 Val Loss: Total - 2.7015, Cost - 2.4369, L2 - 0.2646, Action - 0.54, Start - 0.72, End - 0.64
Epoch-13 Train Loss: Total - 1.5056, Cost - 1.2391, L2 - 0.2665, Action - 0.28, Start - 0.34, End - 0.34
Epoch-13 Val Loss: Total - 2.7573, Cost - 2.4893, L2 - 0.2681, Action - 0.54, Start - 0.73, End - 0.67
Epoch-14 Train Loss: Total - 1.4797, Cost - 1.2095, L2 - 0.2701, Action - 0.27, Start - 0.34, End - 0.33
Epoch-14 Val Loss: Total - 2.7722, Cost - 2.5003, L2 - 0.2719, Action - 0.56, Start - 0.73, End - 0.66
Epoch-15 Train Loss: Total - 1.4609, Cost - 1.1868, L2 - 0.2741, Action - 0.27, Start - 0.33, End - 0.32
Epoch-15 Val Loss: Total - 2.7996, Cost - 2.5237, L2 - 0.2759, Action - 0.55, Start - 0.75, End - 0.67
Epoch-16 Train Loss: Total - 1.4420, Cost - 1.1635, L2 - 0.2785, Action - 0.26, Start - 0.32, End - 0.32
Epoch-16 Val Loss: Total - 2.9070, Cost - 2.6266, L2 - 0.2804, Action - 0.58, Start - 0.79, End - 0.69
Epoch-17 Train Loss: Total - 1.4203, Cost - 1.1376, L2 - 0.2827, Action - 0.26, Start - 0.31, End - 0.31
Epoch-17 Val Loss: Total - 2.8552, Cost - 2.5704, L2 - 0.2848, Action - 0.56, Start - 0.74, End - 0.71
Epoch-18 Train Loss: Total - 1.4002, Cost - 1.1127, L2 - 0.2875, Action - 0.25, Start - 0.31, End - 0.30
Epoch-18 Val Loss: Total - 2.9653, Cost - 2.6757, L2 - 0.2896, Action - 0.58, Start - 0.79, End - 0.72
Epoch-19 Train Loss: Total - 1.3738, Cost - 1.0817, L2 - 0.2921, Action - 0.24, Start - 0.30, End - 0.30
Epoch-19 Val Loss: Total - 2.9388, Cost - 2.6445, L2 - 0.2943, Action - 0.58, Start - 0.78, End - 0.70

**Comment by wzmsltw** [comment]
Yes, the overfitting always happens in TEM training quickly. Thus, actually
I just decay learning rate when loss start overfitting, and take the model
of first epoch after loss decay.

cinjon <notifications@github.com> 于2019年9月12日周四 上午12:38写道：

**Comment by cinjon** [comment]
So you're taking something like the fifth epoch only?

**Comment by wzmsltw** [comment]
@cinjon  Since THUMOS is a small dataset, TEM overfit quickly. In codes I shared, I used epoch-3 (epoch-5 is also good) model to avoid overfitting.
